### DISCUSSION TITLE
another NIC for beats/osquery etc in SO 2.x

### DISCUSSION THREAD

**Discussion by SirLivingman** [Security-Onion-Solutions/securityonion#2312] [Unsupported Versions]
I have a Standalone 2.3.10 SO installation on a physical machine with 4 NIC interfaces.
When setting that up i configured the Management (1nic) and Monitor (1nic) interfaces as specified in the guide.
All of that is working as expected, then i add another NIC to get events from Beats agents (1nic) run so-allow select b and looks OK.
The new NIC is added on the machine with correct IP and so on. However that nic seems impossible to reach with telnet or ping or any other way. When changing the beat agent klient to the management interface it connects, in earlier SO 16.x that worked (havning a separate nic for beats traffic) but not in 2.x is that as expected? Is it supposed to be that beats/osquery/other and management should share the same NIC? or am i missing some configuration here.
I have tested with several servers/klients (so no nic problem) and diffrent nv-cables so no hardware issue.

**Comment by weslambert** [comment]
I haven't tested this with a second NIC.  Is the intention to have the management NIC and the Beats NIC on a separate network, or are you concerned with the throughput?

> I have a Standalone 2.3.10 SO installation on a physical machine with 4 NIC interfaces.
> When setting that up i configured the Management (1nic) and Monitor (1nic) interfaces as specified in the guide.
> All of that is working as expected, then i add another NIC to get events from Beats agents (1nic) run so-allow select b and looks OK.
> The new NIC is added on the machine with correct IP and so on. However that nic seems impossible to reach with telnet or ping or any other way. When changing the beat agent klient to the management interface it connects, in earlier SO 16.x that worked (havning a separate nic for beats traffic) but not in 2.x is that as expected? Is it supposed to be that beats/osquery/other and management should share the same NIC? or am i missing some configuration here.
> I have tested with several servers/klients (so no nic problem) and diffrent nv-cables so no hardware issue.

I haven't tested this with a second NIC.  Is the intention to have the management NIC and the Beats NIC on a separate network, or are you concerned with the throughput?

**Reply by SirLivingman** [reply]
Yes the intention is to have management and Beats (or other agents) on 2 separate NIC, lots of diffrent reasons for that, we had no problem doing that in SO 16.x and have tried the same way in 2.x but seems hard to get that to work, have another colleague that ran into the same problem.

**Reply by weslambert** [reply]
It might be related to the firewall, or how network interfaces are being managed.  I'd have to look into it more.

**Reply by SirLivingman** [reply]
Thanks that would be awesome

**Comment by Paganwishes** [comment]
I'd love to have the option to have a separate management interface. This would mitigate the risk of the **SOC** being inaccessible due to network congestion.

I think there are some things that would need to be configured in order for this to work in the current configuration. 

1. Obviously a static route should be added to the secondary (backend) interface, having the default route on the "frontend" interface makes the most sense I think.
1. Manager `url_base` in **global.sls** pillar should be changed since the webserver will redirect to this value.
2. The secondary interface will be "disabled" as per:

https://github.com/Security-Onion-Solutions/securityonion/blob/bafb13fd6d9f83a4f4af8ee9a79e7034dbf1379a/setup/install_scripts/99-so-checksum-offload-disable#L1-L8

This would need to be solved in a way that is persistent across future version updates.
Maybe the management interface could be defined here?:

 https://github.com/Security-Onion-Solutions/securityonion/blob/bafb13fd6d9f83a4f4af8ee9a79e7034dbf1379a/setup/automation/pm_standalone_defaults#L45-L52

Let's say using a separate var called **MGMTNIC** , this would be something like this:

```bash
#!/bin/bash

if [[ "$DEVICE_IFACE" != "$MNIC" && "$DEVICE_IFACE" != "$MGMTNIC" && "$DEVICE_IFACE" != *"docker"* ]]; then
	for i in rx tx sg tso ufo gso gro lro; do
		ethtool -K "$DEVICE_IFACE" "$i" off;
  	done
	ip link set dev "$DEVICE_IFACE" arp off multicast off allmulticast off promisc on
fi
```

I don't have my environment in front of me right now, but I will do some tests tomorrow when I get back to work.

**Reply by Paganwishes** [reply]
**Quick update:** 
I updated `base_url` in **global.sls** and edited **99-so-checksum-offload-disable** to ignore my "backend" interface, like this (ens224 = backend):

```bash
#!/bin/bash

if [[ "$DEVICE_IFACE" != "ens192" && "$DEVICE_IFACE" != "ens224" && "$DEVICE_IFACE" != *"docker"* ]]; then
	for i in rx tx sg tso ufo gso gro lro; do
		ethtool -K "$DEVICE_IFACE" "$i" off;
  	done
	ip link set dev "$DEVICE_IFACE" arp off multicast off allmulticast off promisc on
fi
```

Now the **SOC** is available on the backend interface. 

> **NOTE**
> I've only tried this on a manager-node in a distributed SO-configuration. This should work in a standalone configuration aswell.
